src/styles/main.scss

```diff
@@ -1,3 +1,80 @@
-body {
-  margin: 0;
+$padding: 10px;
+$gap: 1px;
+$columns: 7;
+$cellSize: 100px;
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: flex-start;
+  padding: $padding;
+  max-width: ($cellSize + $gap) * ($columns + 1);
+  margin: 0 auto;
+  gap: $gap;
+
+  &__day {
+    width: $cellSize;
+    height: $cellSize;
+    background-color: #eee;
+    border: 1px solid black;
+    position: relative;
+    transition: all ease-in-out 0.3s;
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transform: translateY(-20px);
+    }
+
+    &::before {
+      content: "";
+      font-family: Arial, Helvetica, sans-serif;
+      font-size: 30px;
+      position: absolute;
+      top: 50%;
+      left: 50%;
+      transform: translate(-50%, -50%);
+      text-align: center;
+    }
+  }
+
+  $monthDays: (
+    "sun": 0,
+    "mon": 1,
+    "tue": 2,
+    "wed": 3,
+    "thu": 4,
+    "fri": 5,
+    "sat": 6,
+  );
+
+  @each $day, $index in $monthDays {
+    &--start-day-#{$day} &__day {
+      &:first-child {
+        margin-left: ($cellSize + $gap * 2.75) * $index;
+      }
+    }
+  }
+
+  $monthLength: 28, 29, 30, 31;
+
+  @each $length in $monthLength {
+    &--month-length-#{$length} &__day {
+      &:nth-child(n + #{$length + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  @for $i from 1 through 31 {
+    &--month-length-31 .calendar__day:nth-child(#{$i})::before,
+    &--month-length-30
+    .calendar__day:nth-child(-n + 30):nth-child(#{$i})::before,
+    &--month-length-29
+    .calendar__day:nth-child(-n + 29):nth-child(#{$i})::before,
+    &--month-length-28
+    .calendar__day:nth-child(-n + 28):nth-child(#{$i})::before {
+      content: "#{$i}";
+    }
+  }
```

Comment:
You can make it way simpler. As a hint I can tell you that you should do this loop inside of `__day` and you need only one selector inside of it